src/header/mod.rs

```diff
@@ -320,10 +320,11 @@ impl Headers {
     }
 
     /// Removes a header from the map, if one existed.
-    /// Returns true if a header has been removed.
-    pub fn remove<H: Header>(&mut self) -> bool {
+    /// Returns the header, if one has been removed.
```

Comment:
I can't think of a reason why `Raw` is needed, but with the current return type there is no upgrade path for people who were checking the boolean value, as `.is_some()` is `false` if the header *was* removed, but could not be parsed.